api/v1beta1/validator.go

```diff
@@ -30,7 +47,121 @@ func validOcid(ocid string) bool {
 }
 
 // validShape is a simple pre-flight
-// we will let the serverside handle the more complex and compete validation
+// we will let the serverside handle the more complex and compete validation.
 func validShape(shape string) bool {
 	return len(shape) > 0
 }
+
+// validRegion test if the string can be a region.
+func validRegion(stringRegion string) bool {
+	if blankRegex.MatchString(stringRegion) || stringRegion == "" {
```

Comment:
region can be blnak right? we should return true if it is blank right?

Reply:
In our case a region can be blank. But a valid region can't be blank. 

So the validate function, to me, seems correct. But in the webhook we don't call it if the region is empty. 

I could see it either way it would be less code if we allowed for blank in the validate function 🤷‍♂️

Reply:
hmm yeah, I think the fact that stringRegion == "" check is there seems confusing to me. Can we just do this in validRegions
if blank{
return true
}
// check for invalid region in case of non empty
Since we are not validating much in region anyways, I am ok removing this validation itself?

Reply:
oh yeah I need to rename `blankRegex` 🤦 
it should just be something like `validnameRegex` or something. Right now it is only checking for spaces in strings. I will update this.